shell/common/keyboard_util.cc

```diff
@@ -319,11 +320,17 @@ ui::KeyboardCode KeyboardCodeFromCharCode(char16_t c, bool* shifted) {
   }
 }
 
-ui::KeyboardCode KeyboardCodeFromStr(const std::string& str, bool* shifted) {
-  if (str.size() == 1)
-    return KeyboardCodeFromCharCode(str[0], shifted);
-  else
-    return KeyboardCodeFromKeyIdentifier(str, shifted);
+ui::KeyboardCode KeyboardCodeFromStr(const std::string& str,
+                                     base::Optional<char16_t>* shifted_char) {
```

Comment:
I'm noticing a lot of code for `shifted_char`, but what about characters that require alt (aka option on macOS) to input? For example, in the french keyboard layout, typing `@` requires pressing the option key (at least on macOS). Would this "alternate kind of shifted character" still work with these changes since they assume a US keyboard layout?

Reply:
In Electron apps, the accelerators are written in US keyboard layout, so we only consider the shifted state for the keys.

For macOS, if the accelerator is `@`, the character will be used as it is, and macOS will automatically handle it for different keyboard layouts.

But for Windows/Linux, the `@` accelerator is internally translated as <kbd>Shift</kbd> + <kbd>2</kbd>, and it will show as different keys depending on the actual keyboard layout.

So when using `@` as accelerator, the actual key will be different on macOS and Windows/Linux depending on keyboard layout.